wp-includes/mysql/class-mysql-lexer.php

```diff
@@ -0,0 +1,2752 @@
+	 * Some of the synonyms may have a different keyword/function status and
+	 * version constraints and thus need to be converted at the end.
+	 *
+	 * E.g.: NOW is a non-reserved keyword that needs to be used with "()".
+	 *       CURRENT_TIMESTAMP is a reserved keyword that can be used without "()".
+	 */
+	const SYNONYMS = array(
+		self::CHARACTER_SYMBOL           => self::CHAR_SYMBOL,
+		self::CURRENT_DATE_SYMBOL        => self::CURDATE_SYMBOL,
+		self::CURRENT_TIME_SYMBOL        => self::CURTIME_SYMBOL,
+		self::CURRENT_TIMESTAMP_SYMBOL   => self::NOW_SYMBOL,
+		self::DAYOFMONTH_SYMBOL          => self::DAY_SYMBOL,
+		self::DEC_SYMBOL                 => self::DECIMAL_SYMBOL,
+		self::DISTINCTROW_SYMBOL         => self::DISTINCT_SYMBOL,
+		self::FIELDS_SYMBOL              => self::COLUMNS_SYMBOL,
+		self::FLOAT4_SYMBOL              => self::FLOAT_SYMBOL,
+		self::FLOAT8_SYMBOL              => self::DOUBLE_SYMBOL,
+		self::INT1_SYMBOL                => self::TINYINT_SYMBOL,
+		self::INT2_SYMBOL                => self::SMALLINT_SYMBOL,
+		self::INT3_SYMBOL                => self::MEDIUMINT_SYMBOL,
+		self::INT4_SYMBOL                => self::INT_SYMBOL,
+		self::INT8_SYMBOL                => self::BIGINT_SYMBOL,
+		self::INTEGER_SYMBOL             => self::INT_SYMBOL,
+		self::IO_THREAD_SYMBOL           => self::RELAY_THREAD_SYMBOL,
+		self::LOCALTIME_SYMBOL           => self::NOW_SYMBOL,
+		self::LOCALTIMESTAMP_SYMBOL      => self::NOW_SYMBOL,
+		self::MID_SYMBOL                 => self::SUBSTRING_SYMBOL,
+		self::MIDDLEINT_SYMBOL           => self::MEDIUMINT_SYMBOL,
+		self::NDB_SYMBOL                 => self::NDBCLUSTER_SYMBOL,
+		self::RLIKE_SYMBOL               => self::REGEXP_SYMBOL,
+		self::SCHEMA_SYMBOL              => self::DATABASE_SYMBOL,
+		self::SCHEMAS_SYMBOL             => self::DATABASES_SYMBOL,
+		self::SESSION_USER_SYMBOL        => self::USER_SYMBOL,
+		self::SOME_SYMBOL                => self::ANY_SYMBOL,
+		self::SQL_TSI_DAY_SYMBOL         => self::DAY_SYMBOL,
+		self::SQL_TSI_HOUR_SYMBOL        => self::HOUR_SYMBOL,
+		self::SQL_TSI_MICROSECOND_SYMBOL => self::MICROSECOND_SYMBOL,
+		self::SQL_TSI_MINUTE_SYMBOL      => self::MINUTE_SYMBOL,
+		self::SQL_TSI_MONTH_SYMBOL       => self::MONTH_SYMBOL,
+		self::SQL_TSI_QUARTER_SYMBOL     => self::QUARTER_SYMBOL,
+		self::SQL_TSI_SECOND_SYMBOL      => self::SECOND_SYMBOL,
+		self::SQL_TSI_WEEK_SYMBOL        => self::WEEK_SYMBOL,
+		self::SQL_TSI_YEAR_SYMBOL        => self::YEAR_SYMBOL,
+		self::STDDEV_POP_SYMBOL          => self::STD_SYMBOL,
+		self::STDDEV_SYMBOL              => self::STD_SYMBOL,
+		self::SUBSTR_SYMBOL              => self::SUBSTRING_SYMBOL,
+		self::SYSTEM_USER_SYMBOL         => self::USER_SYMBOL,
+		self::VAR_POP_SYMBOL             => self::VARIANCE_SYMBOL,
+		self::VARCHARACTER_SYMBOL        => self::VARCHAR_SYMBOL,
+	);
+
+	/**
+	 * Version constraints for version-specific tokens.
+	 *
+	 * Positive number: >= <version>
+	 * Negative number: <  <version>
```

Comment:
The range is inclusive for positive numbers and exclusive for negative numbers. What's your take on using either two inclusive ranges (`>=` and `<=`) or two exclusive ranges (`>` and `<`)?

Reply:
@adamziel I kind of like this distinction because it represents `Introduced in <version>` and `Removed in <version` as in the table from MySQL docs: https://dev.mysql.com/doc/mysqld-version-reference/en/keywords.html

I'm improving the doc to explain that.